src/internal/collection/postgres_collection.go

```diff
@@ -301,20 +328,72 @@ func (c *postgresCollection) list(
 		query += " where " + strings.Join(fields, " and ")
 	}
 
-	if opts.Order != SortNone {
-		if order, err := orderToSQL(opts.Order); err != nil {
-			return err
-		} else if target, err := targetToSQL(opts.Target); err != nil {
-			return err
+	afterCondition := func(t SortTarget, ord SortOrder) (string, error) {
+		ts, err := targetToSQL(t)
+		if err != nil {
+			return "", err
+		}
+		afterSortVal, err := sortTargetValue(last, t)
+		if err != nil {
+			return "", err
+		}
+		args = append(args, afterSortVal)
+		// condition ends up including:
+		// where ... (<TARGET> <OP> <VALUE> or (<TARGET> = <VALUE> and key > '<PKEY>'))
+		// This covers the case where we only want to return rows following a certain offset value.
+		// We handle the case where multiple rows have the same sort value (when it isn't sorted by primary key)
+		// by including "or (<TARGET> = <VALUE> and key > '<PKEY>')" and also ordering by Primary Key as a tie breaker
+		cond := fmt.Sprintf("(%s %s $%d or (%s = $%d and key > '%s'))", ts, sortOrderOperator(ord), len(args), ts, len(args), last.Key)
+		if len(withFields) > 0 {
+			return " and " + cond, nil
+		} else {
+			return " where " + cond, nil
+		}
+	}
+
+	st := opts.Target
+	so := opts.Order
+	if opts.Order == SortNone { // defaults if to "order by key asc" if no sort specified
+		st = SortByKey
+		so = SortAscend
+	}
+	if order, err := orderToSQL(so); err != nil {
+		return "", nil, err
+	} else if target, err := targetToSQL(st); err != nil {
+		return "", nil, err
+	} else {
+		if last != nil {
+			ac, err := afterCondition(st, so)
+			if err != nil {
+				return "", nil, err
+			}
+			query += ac
+		}
+		if st == SortByKey {
+			query += fmt.Sprintf(" order by key %s", order)
 		} else {
-			query += fmt.Sprintf(" order by %s %s", target, order)
+			query += fmt.Sprintf(" order by %s %s, key asc", target, order)
 		}
 	}
 
 	if opts.Limit > 0 {
 		query += fmt.Sprintf(" limit %d", opts.Limit)
 	}
+	return query, args, nil
+}
 
+func (c *postgresCollection) list(
+	ctx context.Context,
+	withFields map[string]string,
+	opts *Options,
+	greedy bool,
```

Comment:
Do we need a greedy option anymore? I don't think we do now that we don't hold a transaction open when listing.

Reply:
You're right 👍

Reply:
Based on the initial slack thread, we decided that the "Dirty read semantics" that are exhibited by Lists that use multiple buffers are acceptable for ReadOnly but, ReadWrite may be more risky.